vmdb/app/models/mixins/authentication_mixin.rb

```diff
@@ -46,8 +46,12 @@ def authentication_password_encrypted(type = nil)
     authentication_component(type, :password_encrypted)
   end
 
+  def required_credential_fields(_type)
```

Comment:
UGH @himdel, thanks. I understand.  Sounds like we need an AuthenticationSet that contains many Authentications and each knows what fields are required... passing around type everywhere is weird.

Ok, I'm good with this for now, thanks @himdel